### PR TITLE
ci: bump sage and use setup action

### PR DESCRIPTION
**Pull request by odsod**
Get caching for free.

### CODE REVIEW

File: .sage/go.mod [hunk 1]
```diff
@@ -4,10 +4,5 @@ go 1.17
 
 require (
 	github.com/go-logr/logr v1.2.2
```
Comment (Edholm) [inline .sage/go.mod:6]: this should've been removed?

**Review by Edholm — APPROVED** [review]
woopedoop!

File: .sage/go.mod (hunk 1 above)
Comment (Edholm) [reply .sage/go.mod:6]: also make files need to be updated :)

Comment (odsod) [reply .sage/go.mod]: Aye, fixed!

**Review by Edholm — APPROVED** [review]
:broom:

**Pull request merged**